### DISCUSSION TITLE
Support for CLANG 17 on manylinux2014 (aarch64)

### DISCUSSION THREAD

**Discussion by Laky-64** [pypa/cibuildwheel#1909] [Q&A]
Hello,

I'm in need of using CLANG 17 on the manylinux2014 platform for the aarch64 architecture. This is essential for my project's requirements. Is it possible to add support for CLANG 17 specifically for the aarch64 architecture within the cibuildwheel environment? I would appreciate guidance or any available solutions to configure cibuildwheel accordingly.

Thank you for your consideration and assistance.

**Comment by makslevental** [comment]
This isn't possible because manylinux determines which compilers are supported not cibuildwheel.

**Comment by joerick** [comment]
It's possible to make your own manylinux image and use that via the CIBW_MANYLINUX_IMAGE options. I believe @Czaki does that for his [imagecodecs build](https://github.com/Czaki/imagecodecs_build/blob/master/build_utils/Dockerfile_manylinux2014_x86_64), though i'm not 100% sure of the specifics. 

Alternatively, if the installation is quick, you could add the command in CIBW_BEFORE_ALL to install clang17.

**Comment by Czaki** [comment]
I prebuilt libraries required for imagecodecs to speedup CI time (building libraries take half hour). I do not play with compiler but I think it should be doable. 

It may be not trivial as it needs to link against a proper libraries to pass auditwheel.

**Comment by Laky-64** [comment]
Well, imagecodecs is a separate thing, the problem is to install clang17 more than anything else, it's a difficult challenge to do it
